DuckDuckGo/Favicons.swift

```diff
@@ -35,7 +35,8 @@ public class Favicons {
         static let tabsCache = CacheType.tabs.create()
         static let appUrls = AppUrls()
         static let targetImageSizePoints: CGFloat = 64
-        
+        static let bookmarksCoreDataStorage = BookmarksCoreDataStorage.shared
```

Comment:
I spent a couple of hours on this and it cannot be done without changing the singleton nature of Favicons which I think greatly enhances the scope of this bug fix PR.  Favicons in its current state is not designed for DI.  I would recommend we create this as a new issue (Refactor FavIcons for DI + testability?) since this could cause other regressions in the app (and I don't know all the use cases / edge cases to test)

Reply:
I deleted my last comment and I've made this change. It would be nice in a future issue to remove the singleton so this could be DI'd as I can't use the MockBookmarksCoreDataStore for unit tests

Reply:
I updated my test and all the pre-existing tests to no longer use the singleton